src/formats/smilesformat.cpp

```diff
@@ -2364,6 +2364,57 @@ namespace OpenBabel {
     return(idx);
   }
 
+  // Is this dbl bond in a ring that is of size <= IMPLICIT_CIS_RING_SIZE?
+  //
+  // A bounded BFS from one side of a bond trying to find the other side.
+  //   - The required queue is implemented using a std::vector for efficiency
+  //   - Note that items are never popped from the queue, I just move a pointer
+  //     along to mark the left hand side
+  //   - A potential improvement would be to BFS from both sides at the same time
+  static bool IsNotInRingOfSizeThatImpliesCis(OBMol *mol, OBBond *bond)
+  {
+    if (!bond->IsInRing()) return true;
+    OBAtom* start = bond->GetBeginAtom();
+    OBAtom* end = bond->GetEndAtom();
+    std::vector<OBAtom*> qatoms;
+    unsigned int numatoms = mol->NumAtoms();
+    qatoms.reserve(numatoms < 42 ? numatoms : 42); // Value chosen by inspection on ChEMBL when using maxsize of 8
+    OBBitVec seen(mol->NumAtoms() + 1);
+    seen.SetBitOn(start->GetIdx());
+    FOR_BONDS_OF_ATOM(nbond, start) {
+      if (&*nbond == bond) continue;
+      if (!nbond->IsInRing()) continue;
+      OBAtom* nbr = nbond->GetNbrAtom(start);
+      qatoms.push_back(nbr);
+    }
+    unsigned int depthmarker = qatoms.size(); // when qstart reaches this, increment the depth
+    unsigned int qstart = 0;
+    unsigned int depth = 2;
+    while (qatoms.size() - qstart) { // While queue not empty
+      OBAtom* curr = qatoms[qstart];
+      if (qstart == depthmarker) {
+        depth++;
+        depthmarker = qatoms.size();
+      }
+      qstart++;
+      if (seen.BitIsSet(curr->GetIdx()))
+        continue;
+      seen.SetBitOn(curr->GetIdx());
+      if (depth < IMPLICIT_CIS_RING_SIZE) {
```

Comment:
IMPLICIT_CIS_RING_SIZE should be a method parameter

Reply:
Sure.